pyston/resource.py

```diff
@@ -765,8 +765,16 @@ def render_response(self, result, http_headers, status_code, fieldset):
         return super(BaseObjectResource, self).render_response(result, http_headers, status_code, fieldset)
 
     def _get_allow_header(self):
+        try:
+            obj = self._get_obj_or_none()
+        except Exception:
```

Comment:
I am little bit scary with this. _get_obj_or_none should never raise exception. Therefore maybe you should do this in your code not in the library

Reply:
The problem is I cannot recognize from which method is `_get_obj_or_none()` called. In this case, I want to suppress the exception, but if it is called from "normal" code, I want exception to be passed further and processed by the resource, with help of `exception_responses` attribute, which will convert the exception to the correct response.

I don't understand why `get_obj_or_none` should not raise exception. It is perfectly normal mechanism of reporting error. Just the opposite - this code should never fail, because it will prevent rendering of the response, which is a reaction to an exception.